exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetFormatPlugin.java

```diff
@@ -208,8 +209,18 @@ public DrillTable isReadable(DrillFileSystem fs, FileSelection selection,
         FileSystemPlugin fsPlugin, String storageEngineName, String userName)
         throws IOException {
       // TODO: we only check the first file for directory reading.
-      if(selection.containsDirectories(fs)){
-        if(isDirReadable(fs, selection.getFirstPath(fs))){
+      if(selection.containsDirectories(fs)) {
+        Path dirMetaPath = new Path(selection.getSelectionRoot(), Metadata.METADATA_DIRECTORIES_FILENAME);
+        if (fs.exists(dirMetaPath)) {
+          ParquetTableMetadataDirs mDirs = Metadata.readMetadataDirs(fs, dirMetaPath.toString());
+          if (mDirs.getDirectories().size() > 0) {
+            FileSelection dirSelection = FileSelection.createFromDirectories(mDirs.getDirectories(), selection);
+            dirSelection.setExpandedPartial();
+            return new DynamicDrillTable(fsPlugin, storageEngineName, userName,
```

Comment:
I intentionally don't call isDirReadable() here because that method returns true if a metadata cache file exists and I am doing a similar check for the directories file here with fs.exists(dirMetaPath).    If this check fails,  we will fall through to the old code path (line 223) which does check isDirReadable().